syntax = "proto3";

package image;

message Image {
  uint32 width = 1;
  uint32 height = 2;
  uint32 channel = 3;
  uint32 image_size = 4;
  uint64 timestamp = 5;
  int32 depth_range = 6;
  bytes data = 7;
}

service ImageService{
  rpc SendRGBDImage (ImageRequest) returns (ImageRGBDReply) {}
  rpc SendWDRImage (ImageRequest) returns (ImageWDRReply) {}  
}

message ImageRequest{
}

message ImageRGBDReply{
  Image image_rgb = 1;
  Image image_depth = 2;
}

message ImageWDRReply{
  Image image_depth_range1 = 1;
  Image image_depth_range2 = 2;
}